### ISSUE TITLE
2.5.0b in CentOS6 docker image on CentOS7 host

### ISSUE THREAD

**Issue by chihlee** [alexdobin/STAR#186]
Hello Alex,

I got this error running the static STAR binary inside a CentOS6 docker image on a CentOS7 host:

```
EXITING because of FATAL ERROR: number of bytes expected from the BAM bin does not agree with the actual size on disk: 2038236188 203985875 16
```

It runs on a CentOS6 host without any problem. I was able to run the latest release (2.5.2.b) inside a CentOS6 docker image without any problem on a CentOS6/7 host. This happens only when using more than one thread.

What is the difference between 2.5.0b and 2.5.2b besides the new commits? Was a newer gcc/g++ used to build 2.5.2b? Thanks.

Best,
Steve

**Comment by alexdobin** [comment]
Hi Steve,

I indeed changed the gcc/g++ to 4.7.0 for 2.5.2b, as some users reported incompatibility with 5.3.0 I had been using before. I guess I could provide the static executables compiled with both 4.7.0 and 5.3.0. Do you want to try them out?

Cheers
Alex

**Comment by chihlee** [comment]
Hello Alex,

That would be very helpful! I would love to try them out.

I also built a static STAR binary inside a CentOS7 image using gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-4). However, this one crashes when running inside a CentOS6 docker image on a CentOS7 host as well.

Best,
Steve

**Comment by alexdobin** [comment]
Hi Steve,

I have added gcc 4.7.0 and 5.3.0 executables to the master on GitHub. Please let me know whether it fixes the problem, and I will make a tagged release.

Cheers
Alex

**Comment by chihlee** [comment]
Hello Alex,

Thanks for adding the executables. The current version (2.5.2b) runs without any problem inside docker. We are experiencing this issue with version 2.5.0b. Is there a way to get them for version 2.5.0b? Thanks.

Best,
Steve

**Comment by alexdobin** [comment]
Hi Steve,

sorry, I thought the problem was with the later release.
I have added the gcc 4.7.0 and 5.3.0 executables here:
[https://github.com/alexdobin/STAR/releases/download/2.5.0b/Linux_x86_64_static_gcc.tgz](https://github.com/alexdobin/STAR/releases/download/2.5.0b/Linux_x86_64_static_gcc.tgz
)
I do not want to change the 2.5.0b tag, but, if it helps, I can make a 2.5.0b.1 tag.

Cheers
Alex

**Comment by chihlee** [comment]
Hello Alex,

No worries. Thank you so much! We'll test it out. No need to make a new tag.

Best,
Steve

**Comment by chihlee** [comment]
Hello Alex,

We tested the executables inside docker but STAR crashes still. Not sure why but probably a bug that's triggered by running inside docker on a CentOS7 host. The latest version 2.5.2b works great. So whatever the bug is, it's likely fixed in 2.5.2b. We've decided to move on with 2.5.2b.

Please feel free to close the issue. Thanks again for all your help.

Best,
Steve
